persistence/rocksdb/src/main/java/org/infinispan/persistence/rocksdb/RocksDBStore.java

```diff
@@ -358,6 +368,34 @@ public MarshalledEntry load(Object key) {
         }
     }
 
+    @Override
+    public void writeBatch(List<MarshalledEntry<? extends K, ? extends V>> marshalledEntries) {
+        try {
+            WriteBatch writeBatch = new WriteBatch();
```

Comment:
I know that you didn't write this originally, but it looks like the expiredDb updates in `purge` method are badly in need of this same exact batching.

Also not to mention it scares me that you can only insert 10000 elements into the rocks db store until you will block a thread until the expiration reaper is ran. This needs to be cleaned up @tristantarrant . I would say instead of calling put at https://github.com/infinispan/infinispan/blob/master/persistence/rocksdb/src/main/java/org/infinispan/persistence/rocksdb/RocksDBStore.java#L478 we should invoke offer and if we weren't able to insert to start off a thread to clear this out or do it inline.